pennylane/capture/dynamic_shapes.py

```diff
@@ -113,3 +115,82 @@ def f(n):
         return None, ()
     abstracted_axes = jax.tree_util.tree_unflatten(structure, abstracted_axes)
     return abstracted_axes, abstract_shapes
+
+
+def register_custom_staging_rule(
+    primitive, get_outvars_from_params: Callable[[dict], list["jax.core.Var"]]
+) -> None:
+    """Register a custom staging rule for a primitive, where the output should match the variables retrieved by
+    ``get_outvars_from_params``.
+
+    Args:
+        primitive (jax.core.Primitive): a jax primitive we want to register a custom staging rule for
+        get_outvars_from_params (Callable[[dict], list[jax.core.Var]]): A function that takes in the equation's ``params``
+            and returns ``jax.core.Var`` we need to mimic for the primitives return.
+
+    For example, the ``cond_prim`` will request its custom staging rule like:
+
+    .. code-block:: python
+
+        register_custom_staging_rule(cond_prim, lambda params: params['jaxpr_branches'][0].outvars)
```

Comment:
I guess the purpose of this code block is to show how the function should be called

```suggestion
    .. code-block:: python
    
        from pennylane.capture import register_custom_staging_rule
        from pennylane.capture.primitives import cond_prim
        register_custom_staging_rule(cond_prim, lambda params: params['jaxpr_branches'][0].outvars)
```

Reply:
I think having the imports might encourage users to run the code, where as this is to demonstrate how it might be used in source code.  I can add a warning saying that this should just be used by developers in source code?

Reply:
Thanks for the suggestion, but I'm going to leave them out since this is showing how to use it in source code, not something to get dynamically used at any point.